ReactNativeClient/lib/components/screens/note.js

```diff
@@ -342,6 +346,29 @@ class NoteScreenComponent extends BaseScreenComponent {
 		await shared.saveOneProperty(this, name, value);
 	}
 
+	undoButton_onPress() {
+		this.setState({
+			noteArr: this.state.note.body.split(' '),
+		}, () => {
+			const noteUndo = this.state.noteArr.pop();
+			this.setState({
+				noteArrHistory: this.state.noteArrHistory.concat(noteUndo),
+				note: { body: this.state.noteArr.join(' ') },
+			});
+		});
+	}
+
+	redoButton_onPress() {
+		const noteRedo = this.state.noteArrHistory.pop();
+		this.setState({
+			noteArr: this.state.noteArr.concat(noteRedo),
+		}, () => {
+			this.setState({
+				note: { body: this.state.noteArr.join(' ') },
+			});
+		});
+	}
```

Comment:
What happens if I put the cursor at the beginning of a note, write down something and press Undo? And how about languages like Chinese that have almost no spaces?

Reply:
Implementing undo/redo is not impossible but it can't be done with a quick hack like this.

Reply:
I understand your concern let me check what can be done

Reply:
@laurent22 I have tested the undo/redo using the timeline library https://github.com/uditalias/timeline it works great also the size of the library is less than 1 KB. Let me know if I can make the changes

Reply:
> https://github.com/uditalias/timeline

how well know is it? any references, just thinking about broken dependencies. NPM stats are quite low https://www.npmjs.com/package/@udidu/timeline

Reply:
> https://github.com/uditalias/timeline

is that already part of the code?
Otherwise you may to a PR using this library.

Reply:
@PackElend I have written a custom UndoRedoService for the app and am testing it so it works perfectly in all the cases. I will send the PR soon, though this PR has been closed I will make a new PR ASAP once the testing is done.

Reply:
also drop a note in regard of my dependencies concerns in the new PR

Reply:
@PackElend am not using the library I made my own service which does undo-redo perfectly. I will send the PR in the next few hours